### ISSUE TITLE
Add the ability to transfer books to a kindle.

### ISSUE THREAD

**Issue by Hedronmx** [babluboy/bookworm#133]
This would make me migrate from Calibre to Bookworm :D!

**Comment by babluboy** [comment]
@Hedronmx Thanks for this, while format conversion was not on my original plan for Bookworm, this is a good thought to stow away until the time I have nothing else to do on bookworm and own a kindle as well :-). 

At the moment I think I have a long way to go in order to make Bookworm more feature rich, better rendering (especially PDFs) and then think of integration with online book sources and devices (i.e. Kindle)

I am guessing Calibre must be converting the book into .MOBI format and transferring the same into Kindle with some kind of file transfer integration.

**Comment by Hedronmx** [comment]
Well I really don't mind converting to other formats, since I download most my books as mobi... Still it would be a nice feature.

Well you can just drag and drop files into kindle's main memory. I guess Calibre it's just doing that.

I'll sure be waiting for device management ❗️

**Comment by natanielcz** [comment]
Maybe an easier option would be sending mobi/pdf (any format) via email to Kindle email address. And the effect is almost the same (the only thing is that you have to download it on your Kindle at any moment.

**Comment by babluboy** [comment]
@natanielcz Good idea. I didnt know that kindle have a device specific email address. Obviously it has to depend on the user providing the correct kindle address. Sending to an email also has the advantage that the kindle does not require to be turned on and connected to the same network/bluetooth. 
While email is an easy thing to do, probably it will be good to pack the eBook into a format that works well on most kindle versions - i'm guessing .mobi

**Comment by natanielcz** [comment]
@babluboy you can see/modify your device email address on the Amazon website. Just navigate to: `Your Content and Devices` -> `Devices` and click on 3 dots preceding device name. The popover will appear with the device email address. To make it work you also have to add your email address to trusted somewhere on the Amazon website.

In the beginning, It would be great to add the possibility of sending a book via email. Then you may add as many conversions as you want or need :) 

I don't know how to develop Linux apps but I may try to help you with this app If You could provide me some support :)

**Comment by babluboy** [comment]
@natanielcz Many thanks for the research. I will try this today to email a book to my kindle device.
The key will be to pack the book contents into .MOBI format before the same can be sent on email. I have a python script which unpacks .MOBI files, I will check what is required to do the reverse.

Good to know your interest in helping me with Bookworm (at the moment i'm a bit tied up with my day job to look at some of the open issues/feature requests). If you have knowledge in HTML/Javascript then you might help me to the pagination work - currently after unpacking the book I'm displaying the content by the HTML files which are chapters. If a bit of javascript/css can be applied then only the content that fits into the viewport should be displayed. I have seen css scripts for this but not able to find one which works. Taking a long HTML file and applying the script to show part of it with links to show the rest progressively should do the trick. So if you know your way in HTML/CSS/Java script then you can work on it and not need to understand the bookworm code. Happy to guide you

**Comment by babluboy** [comment]
Issue #170 tracks the discussion/effort for pagination

**Comment by natanielcz** [comment]
Thanks. I will take a look at this :)
